platform/ios/src/MGLAnnotationView.h

```diff
@@ -16,6 +18,14 @@ NS_ASSUME_NONNULL_BEGIN
 - (instancetype)initWithReuseIdentifier:(nullable NSString *)reuseIdentifier;
 
 /**
+ This property will be set to the associated annotation when the view is visible.
+ 
+ When the view is queued and waiting to be reused, the value will be set to nil.
+ This property should not be changed directly.
+ */
+@property (nonatomic, strong, nullable) id <MGLAnnotation> annotation;
```

Comment:
It does not look like there is a good reason for us to use the MK approach of warning about changing this property in the comment but not making it readonly. Let's go ahead and make it readonly.